extension/README.md.args.mjs

```diff
@@ -42,110 +36,342 @@ yarn start
+## Checkpoint 1: 🤝 Contributing 💵
+
+>Let's start by implementing a state variable that we will need in the function logic.
+
+⚖️ You'll need to track individual \`balances\` using a mapping. This way we will know who gave what in the event that the funding effort fails to raise enough and everyone needs to be refunded. Add it under the existing fundingRecipient variable.
 
 \`\`\`solidity
-mapping ( address => uint256 ) public balances;
+mapping(address => uint256) public balances;
 \`\`\`
 
-And also track a constant \`threshold\` at \`1 ether\`
+> Next let's add an event. Events are useful for outside services that are watching the chain  for certain things to occur. In our case, the front end is going to use this event to know when a contribution takes place.
+
+📣 Add an event to the contract called \`Contribution\` that receives the address of the contributor and the amount they contributed.
 
 \`\`\`solidity
-uint256 public constant threshold = 1 ether;
+event Contribution(address, uint256);
+\`\`\`
+
+ >📝 Note: If you use named arguments in your event (e.g. \`event Contribution(address indexed contributor, uint256 amount)\`), you'll need to update \`/packages/nextjs/app/contributions/page.tsx\` to reference event parameters by their names instead of numeric indices.
+
+### Implementing the \`contribute()\` function
+
+> 👩‍💻 Now focus on writing your \`contribute()\` function. The payable method already exists but is empty. Go fill it with logic!
+
+The goal of this function is to allow anyone to contribute to the pool of funds. To do this effectively it will need to do the following:
+- Update the \`balances\` mapping
+- Emit the \`Contribute\` event 
+
+<details markdown='1'>
+<summary>🔎 Hint</summary>
+
+You can set mappings like you would access a Javascript array.
+For a mapping like this \`mapping(address => uint256) public dir\` you would access is like this:
+\`\`\`solidity
+address addr = 0x1234...5678;
+dir[addr] += 1;
+\`\`\`
+You need to use the address for the sender of the transaction and you will need to know how much value was sent. Is there an easy way to access these details about the transaction \`msg\`? 🤔 
+
+Go check https://solidity-by-example.org/ if you need help on the syntax.
+
+<details markdown='1'>
+
+<summary>🎯 Solution</summary>
+
+\`\`\`solidity
+function contribute() public payable {
+	balances[msg.sender] += msg.value;
+	emit Contribution(msg.sender, msg.value);
+}
 \`\`\`
 
-> 👩‍💻 Write your \`stake()\` function and test it with the \`Debug Contracts\` tab in the frontend.
+</details>
+</details>
+
+### Try it out
+
+👩‍💻 Now redeploy (\`yarn deploy\`) and go test your function using the \`Debug Contracts\` tab in the front end.
 
-![debugContracts](https://github.com/scaffold-eth/se-2-challenges/assets/55535804/1a888e31-a79b-49ef-9848-357c5cee445a)
+![debugContracts](https://github.com/user-attachments/assets/7d50245b-72f5-433e-b327-0c7e70e83e51)
```

Comment:
note: for me it looks like this now

<img width="1154" height="448" alt="Image" src="https://github.com/user-attachments/assets/1682b7ef-2883-45ad-8d91-72486923397d" />

We'll fix scaffold-ui vars in next PRs (for all the challenges) or you can try to do it like in css args file here https://github.com/scaffold-eth/se-2-challenges/pull/380/changes#diff-2b7f432450f126dd9172bc95baac0faa957d1b81f059551669eb0e8ba4f5df22 . I think just copy-paste should work